### PR TITLE
Track Broker Properties status

### LINKED ISSUE #345: Check properties update status
**Is your feature request related to a problem? Please describe.**
The broker.properties sha is exposed in Jolokia. The fact that the sha in the properties match the generated sha from the defined properties means that the broker is in sync.

**Describe the solution you'd like**
A condition should be used to reflect these states. The operator creates a configMap containing the properties that the broker will read, that means the broker has to be synced. The broker will periodically query the Jolokia endpoint until the broker status finally shows the matching sha from the status. After that, the condition will be updated.

Depends on https://github.com/artemiscloud/activemq-artemis-management/issues/4

### PR DESCRIPTION
**Pull request by ruromero**
Fix #345

### CODE REVIEW

File: api/v1beta1/activemqartemis_types.go [hunk 1]
```diff
@@ -412,4 +412,11 @@ const (
 	DeployedConditionNotReadyReason  = "PodsNotReady"
 	DeployedConditionZeroSizeReason  = "ZeroSizeDeployment"
 	DeployedConditionZeroSizeMessage = "Pods not scheduled. Deployment size is 0"
+
+	BrokerPropsConditionType                             = "BrokerProperties"
```
Comment (gtully) [inline api/v1beta1/activemqartemis_types.go:416]: naming is hard!

I think maybe the condition is BrokerPropertiesInSync or BrokerPropertiesApplied, actually think Applied will help in the future when the status contains errors on failure to apply.

Applied = True -> in sync and no errors

Comment (gtully) [reply api/v1beta1/activemqartemis_types.go:416]: Applied = Unknown for older broker that does not expose Status or when no jolokia agent yet ready

Comment (gtully) [reply api/v1beta1/activemqartemis_types.go:416]: I think any Unknown should filter up to the  (total) Ready condition being unknown

Comment (ruromero) [reply api/v1beta1/activemqartemis_types.go]: Changes applied, I am having some issues with the IT, don't merge yet

**Review by gtully — COMMENTED** [review]
that error re queue struct is a nice enhancement

**Review by gtully — APPROVED** [review]

**Pull request merged**